console/console-init/ui/src/graphql-module/queries/iot_device.ts

```diff
@@ -134,11 +134,24 @@ const UPDATE_IOT_DEVICE = gql(
   }`
 );
 
+const RETURN_ALL_DEVICES_COUNT_FOR_IOT_PROJECT = (iotproject: string) => {
```

Comment:
we have already query for getting device i.e. RETURN_ALL_DEVICES_FOR_IOT_PROJECT.
Only difference is schema which can be dynamic. You can try to update existing query RETURN_ALL_DEVICES_FOR_IOT_PROJECT.
 Modify query:
const RETURN_ALL_DEVICES_FOR_IOT_PROJECT = (
  iotproject: string,
  sortBy?: ISortBy,
  filterObj?: IDeviceFilter,
  resolver?:any
) => {
    const defaultResolver=`
    total
        devices {
          deviceId
          jsonData
          enabled
          viaGateway
        }`;

   if(!resolver){
    resolver=defaultSchema;
   }

  let filter = FILTER_RETURN_ALL_DEVICES_FOR_IOT_PROJECT(
    filterObj || getInitialFilter()
  );

  let orderBy = SORT_RETURN_ALL_DEVICES_FOR_IOT_PROJECT(sortBy);

  const ALL_DEVICE_LIST = gql(
    `query devices_for_iot_project {
      devices(iotproject: "${iotproject}", orderBy:"${orderBy}", filter: "${filter}") {
        ${resolver}
      }
    }`
  );

Note if you will not pass schema it will take default and you need some specfic schema then pass as props . based on that  it will get the data.

const  resolver=`total`;
Note : here as yiu need only total so you can pass schema accordingly. It will get total value only.
RETURN_ALL_DEVICES_FOR_IOT_PROJECT(iotproject,sortBy,filterObj,resolver);

By doing this we can avoid to write so many queries just for different schema.
Feel free to discuss with me if any query.

Reply:
This was helpful. Have implemented it.